src/main.ts

```diff
@@ -1,114 +1,65 @@
 import * as fs from 'fs'
 import * as cp from 'child_process'
 import * as core from '@actions/core'
-import * as toolCache from '@actions/tool-cache'
-import path from 'path'
+import * as path from 'path'
 
 async function run(): Promise<void> {
   try {
+    const localAppDataPath = process.env.LocalAppData
     // get version number from input
     const version = core.getInput('version')
 
-    // get architecture and os
-    const architecture = process.arch
+    // get os
     const os = process.platform
 
-    // map for different platform and arch
-    const extensionMap = {
-      linux: '.tar.gz',
-      darwin: '.zip',
-      win32: '.zip'
+    let windowsInstallScript = `powershell -ex AllSigned -c "Invoke-RestMethod 'https://aka.ms/install-azd.ps1' | Invoke-Expression"`
+    let linuxOrMacOSInstallScript = `curl -fsSL https://aka.ms/install-azd.sh | sudo bash`
+    if (version !== 'latest') {
+      windowsInstallScript = `powershell -ex AllSigned -c "Invoke-RestMethod 'https://aka.ms/install-azd.ps1' -OutFile 'install-azd.ps1'; powershell -ExecutionPolicy Bypass -File ./install-azd.ps1 -Version '${version}'"`
+      linuxOrMacOSInstallScript = `curl -fsSL https://aka.ms/install-azd.sh | sudo bash -s -- --version ${version}`
```

Comment:
Install scripts can also be invoked with version `latest` so you don't need to do extra work here... I think this works:  

```suggestion 
    const windowsInstallScript = `powershell -c "$scriptPath = \"$($env:TEMP)\install-azd.ps1\"; Invoke-RestMethod 'https://aka.ms/install-azd.ps1' -OutFile $scriptPath; . $scriptPath -Version '${latest}'; Remove-Item $scriptPath"`
    const linxOrMacOSInstallScript = `curl -fsSL https://aka.ms/install-azd.sh | sudo bash -s -- --version ${version}`
```